lib/vcs.ml

```diff
@@ -433,9 +410,13 @@ let submodule_update ~dry_run r =
   | (`Git, _, _) as r -> git_submodule_update ~dry_run r
   | `Hg, _, _ -> R.error_msgf "submodule update is not supported with mercurial"
 
-let sanitize_tag = function
-  | `Git, _, _ -> git_sanitize_tag
-  | `Hg, _, _ -> hg_sanitize_tag
+let escape_tag = function
+  | `Git, _, _ -> git_escape_tag
+  | `Hg, _, _ -> hg_escape_tag
+
+let unescape_tag = function
+  | `Git, _, _ -> git_unescape_tag
+  | `Hg, _, _ -> fun x -> x
```

Comment:
I think it's ok to remove the intermediate function but it's propably worth mentioning that `~` is a valid mercurial tag character then!

It would probably look better to have that comment in mercurial specific function but both work for me!

Reply:
I finally looked at what [Mercurial tags](https://www.mercurial-scm.org/wiki/Tag) support and decided to implement it properly. Or, well, close enough to be useful and basically analogous to Git.